src/parse_execution.rs

```diff
@@ -1672,6 +1672,8 @@ impl<'a> ParseExecutionContext {
             job.internal_job_id,
         );
 
+        // Start the timer here to ensure command substitution is measured
+        let _timer = push_timer(job.wants_timing());
```

Comment:
the behavior change looks good (bash does the same) although I think we should consolidate the two calls to `push_timer()` into one. I don't think there is a reason to treat "simple blocks" specially here.
Then we can also get rid of `JobProperties::wants_timing`.
Once we support timing background functions we would want to add something similar back

Reply:
seems fair, I made some changes, but it feels a little weird.

it maintains the current behavior of `fish -c "time begin; echo -n (sleep 1); end &"` returning timing even though timing a background job is an error, since the block isn't run as a background job

not sure if that should be an error too? technically it doesn't do anything wrong since the block isn't backgrounded, but it feels weird syntactically for that to be accepted

Reply:
from what i understand, the timing background rule seems to not apply to blocks?

Reply:
`begin; end &` is not supported yet and the `&` is simply ignored. So it's fine if the `time` behavior is weird.

Reply:
Okay sure, in that case I think the PR is ready for another review, I don't believe the changes I made differ in behavior for the `begin; end &` situation